src/gap-skipper.js

```diff
@@ -154,8 +208,23 @@ export default class GapSkipper {
     let currentTime = this.tech_.currentTime();
     let nextRange = Ranges.findNextRange(buffered, currentTime);
 
-    if (nextRange.length === 0 ||
-        this.timer_ !== null) {
+    if (this.timer_ !== null) {
+      return;
+    }
+
+    if (nextRange.length === 0) {
+      // Even if there is no available next range, there is still a possibility we are
+      // stuck in a gap due to video underflow.
+      let gap = this.gapFromVideoUnderflow_(buffered, currentTime);
+
+      if (gap) {
+        this.logger_('setTimer_:',
+                     'Encountered a gap in video',
+                     'from: ', gap.start,
+                     'to: ', gap.end,
+                     'seeking to current time: ', currentTime);
+        this.tech_.setCurrentTime(currentTime);
```

Comment:
This seems worth a comment about why seeking to `currentTime` resolves this issue.